### ISSUE TITLE
String types const correctness 

### ISSUE THREAD

**Issue by ajkhoury** [ufrisk/MemProcFS#269]
One minor annoyance that I ran into developing with VMMDLL was the lack of string const correctness. One example is the `VMMDLL_ProcessGetProcAddressU` routine, which has the following prototype: 
```
ULONG64 VMMDLL_ProcessGetProcAddressU(_In_ VMM_HANDLE hVMM, _In_ DWORD dwPID, _In_ LPSTR  uszModuleName, _In_ LPSTR szFunctionName);
```
The `uszModuleName` and `szFunctionName` strings are both marked as mutable via the `LPSTR` type, when the strings are never needed to be written to in the internal implementation. This results in a `-Wwrite-strings` warnings when simply calling it like `VMMDLL_ProcessGetProcAddressU(hVMM, 4, "ntoskrnl.exe", "KeGetCurrentIrql")` like you'd normally expect to be able to without warnings. I am proposing [this patch](https://github.com/ufrisk/MemProcFS/files/14440162/memprocfs_const_strings.txt) to add proper const correctness for string types to fix this problem:
[memprocfs_const_strings.txt](https://github.com/ufrisk/MemProcFS/files/14440162/memprocfs_const_strings.txt)

**Comment by ufrisk** [comment]
Thanks,

I'll have to think about this one. And to run some tests on how it will behave with regards to warnings, primarily on Windows, before I decide upon it.

The library as such is fairly popular nowdays and I'm a bit careful to make changes to the external API.

If it seems to be mostly fine I'll look into updating the API.

**Comment by ajkhoury** [comment]
Okay, thank you. I did make the changes so it shouldn't create warnings on Windows, but I did not test on Windows. I will test on Windows tomorrow.

**Comment by ufrisk** [comment]
I have looked into this some more and decided to add this (or a variation of it since I'm not too keen on applying massive patches so I did it manually) in version 4.9.

Users will only get some warnings if using callback methods (vfs* functionality etc.) and that should be fairly rare so I hope it shouldn't cause too much troubles.

The update is probably a week or two away though.

**Comment by ufrisk** [comment]
I've released version 5.9 just about now. I've changed quite a few of these and it should be all good in the external API.

Can you verify it works for you, and if so close the issue.

**Comment by ufrisk** [comment]
I'm closing this issue since I assume everything is working as intended in the latest 5.9 release. Please let me know if you should run into any remaining issues.
